test/test_sparse_semi_structured.py

```diff
@@ -209,18 +213,84 @@ def test_unsupported_dtype(self, dtype, device):
 
         if dtype not in SEMI_STRUCTURED_SUPPORTED_DTYPES:
             with self.assertRaisesRegex(RuntimeError, "Error original_tensor.dtype"):
-                A_sparse = to_sparse_semi_structured(A, mask=A.bool())
+                A_sparse = SparseSemiStructuredTensor(A)
         else:
-            A_sparse = to_sparse_semi_structured(A, mask=A.bool())
+            A_sparse = SparseSemiStructuredTensor(A)
 
     @unittest.skipIf(not _IS_SM8X, "semi-structured sparsity not supported on this library version")
     def test_unsupported_dim(self, device):
         A = torch.rand(128, 128, 128, device=device, dtype=torch.float16)
 
         with self.assertRaisesRegex(RuntimeError, "Error original_tensor.dim"):
-            A_sparse = to_sparse_semi_structured(A, mask=A.bool())
+            A_sparse = SparseSemiStructuredTensor(A)
+
+    @unittest.skipIf(not _IS_SM8X, "semi-structured sparsity not supported on this library version")
+    @unittest.skipIf(TEST_WITH_ROCM, "ROCm doesn't support CUTLASS")
+    @dtypes(*SEMI_STRUCTURED_SUPPORTED_DTYPES)
+    def test_linear_cutlass(self, device, dtype):
+        def run_test(batch_shape, m, n, k, device, dtype, dtype_out, add_bias, activation, rtol, atol):
+            weight, mask = rand_dense_2by4(m, k, dtype, device)
+            input = make_tensor((*batch_shape, n, k), dtype=dtype, device=device)
+            bias = make_tensor((m,), dtype=dtype_out, device=device) if add_bias else None
+
+            dtype_dense = torch.float
+            input_dense = input.to(dtype_dense)
+            weight_dense = weight.to(dtype_dense)
+            bias_dense = bias.to(dtype_dense) if add_bias else None
+            output0 = torch.nn.functional.linear(input_dense, weight_dense, bias=bias_dense)
+            if activation == "relu":
+                relu = torch.nn.ReLU()
+                output0 = relu(output0)
+            elif activation == "silu":
+                silu = torch.nn.SiLU()
+                output0 = silu(output0)
+
+            weight_sparse = weight.masked_select(weight != 0).view(m, k // 2)
+
+            output1, meta = torch._structured_sparse_linear(input, weight_sparse, mask, bias=bias, activation=activation)
+            torch.testing.assert_close(output1.to(dtype_dense), output0, rtol=rtol, atol=atol)
+
+            output1, _ = torch._structured_sparse_linear(input, weight_sparse, meta, bias=bias, activation=activation)
+            torch.testing.assert_close(output1.to(dtype_dense), output0, rtol=rtol, atol=atol)
+
+        is_sm8x = torch.cuda.get_device_capability(0)[0] == 8
+        if not is_sm8x:
+            return
```

Comment:
As the test is decorated with `unittest.skipIf(not _IS_SM8X)` this early exit should never be hit, correct?

Reply:
Correct, removed.